From: Enrico Tassi <gareuselesinge@debian.org>
Date: Fri, 14 Aug 2015 16:11:50 +0200
Subject: consider Hurd as a POSIX system

---
 src/Makefile  | 3 +++
 src/lj_arch.h | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/src/Makefile b/src/Makefile
index f56465d..8a38efd 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -344,6 +344,9 @@ else
   ifeq (GNU/kFreeBSD,$(TARGET_SYS))
     TARGET_XLIBS+= -ldl
   endif
+  ifeq (GNU,$(TARGET_SYS))
+    TARGET_XLIBS+= -ldl
+  endif
 endif
 endif
 endif
diff --git a/src/lj_arch.h b/src/lj_arch.h
index c8d7138..d609b37 100644
--- a/src/lj_arch.h
+++ b/src/lj_arch.h
@@ -79,6 +79,8 @@
 #elif defined(__CYGWIN__)
 #define LJ_TARGET_CYGWIN	1
 #define LUAJIT_OS	LUAJIT_OS_POSIX
+#elif defined(__GNU__)
+#define LUAJIT_OS	LUAJIT_OS_POSIX
 #else
 #define LUAJIT_OS	LUAJIT_OS_OTHER
 #endif
